### DISCUSSION TITLE
Support for transformers

### DISCUSSION THREAD

**Discussion by StellaAthena** [Trusted-AI/adversarial-robustness-toolbox#1282] [Q&A]
Does this repository work with transformers? Especially large autoregressive ones?

**Comment by beat-buesser** [comment]
Hi @StellaAthena Thank you very much for your interest in ART! Yes, it should, but it depends on the task of the transformer if existing estimators can be used or if new estimators would have to be implemented (with little effort). Can you describe your project in more detail?

**Reply by StellaAthena** [reply]
I train large language models, currently on the 1B - 10B scale. I’m looking to experiment with various adversarial attacks against trained transformers operating as language models (be t token prediction), fine-tuned ones for specific tasks, and language models being used for specific tasks via prompt programming.

Right now this is rather exploratory, but what drew me to your library was the wide variety of attacks you have for offer. This would greatly improve my ability to explore the weaknesses of language models at scale by iterating across many different attacks without having to implement them from scratch.

For starters, I will be using the GPT-Neo and GPT-J models via the HuggingFace API. Understanding how to hook up a model in the `transformers` library to this code would be an excellent first step.

**Reply by beat-buesser** [reply]
This sounds very interesting. So far ART has mainly focussed on continuous inputs (e.g. images, video, audio, tables, etc.) and therefore most tools are optimised for these input. We are currently considering better support for discrete inputs like text etc. for all four threats covered by ART but this project is still in an early stage.
For the combination of GPT* and HuggingFace API, have you already considered [TextAttack](https://github.com/QData/TextAttack) focusing on attacks in NLP?
